storage/redis_cluster.go

```diff
@@ -425,66 +284,19 @@ func (r *RedisCluster) GetKeys(filter string) []string {
 // GetKeysAndValuesWithFilter will return all keys and their values with a filter
 func (r *RedisCluster) GetKeysAndValuesWithFilter(filter string) map[string]string {
 	if err := r.up(); err != nil {
-		log.Debug(err)
-		return nil
-	}
-	keys := r.GetKeys(filter)
-	if keys == nil {
-		log.Error("Error trying to get filtered client keys")
 		return nil
 	}
 
-	if len(keys) == 0 {
-		return nil
-	}
-
-	for i, v := range keys {
-		keys[i] = r.KeyPrefix + v
-	}
-
 	client := r.singleton()
-	values := make([]string, 0)
-
-	switch v := client.(type) {
-	case *redis.ClusterClient:
-		{
-			getCmds := make([]*redis.StringCmd, 0)
-			pipe := v.Pipeline()
-			for _, key := range keys {
-				getCmds = append(getCmds, pipe.Get(r.RedisController.ctx, key))
-			}
-			_, err := pipe.Exec(r.RedisController.ctx)
-			if err != nil && err != redis.Nil {
-				log.Error("Error trying to get client keys: ", err)
-				return nil
-			}
-
-			for _, cmd := range getCmds {
-				values = append(values, cmd.Val())
-			}
-		}
-	case *redis.Client:
-		{
-			result, err := v.MGet(r.RedisController.ctx, keys...).Result()
-			if err != nil {
-				log.Error("Error trying to get client keys: ", err)
-				return nil
-			}
-
-			for _, val := range result {
-				strVal := fmt.Sprint(val)
-				if strVal == "<nil>" {
-					strVal = ""
-				}
-				values = append(values, strVal)
-			}
-		}
+	values, err := client.GetKeysAndValuesWithFilter(r.context(), filter)
+	r.check(err, "GetKeysAndValuesWithFilter", filter)
+	if err != nil || len(values) == 0 {
+		return nil
 	}
 
 	m := make(map[string]string)
-	for i, v := range keys {
-		m[r.cleanKey(v)] = values[i]
```

Comment:
what happens with `m[r.cleanKey(v)]`? I think we are omitting the prefixing adds overall